wallet/wallet.c

```diff
@@ -2594,6 +2611,8 @@ const struct forwarding *wallet_forwarded_payments_get(struct wallet *w,
 
 		cur->channel_in.u64 = sqlite3_column_int64(stmt, 4);
 		cur->channel_out.u64 = sqlite3_column_int64(stmt, 5);
+		cur->received_time = sqlite3_column_int64(stmt, 6);
+		cur->resolved_time = sqlite3_column_int64(stmt, 7);
```

Comment:
Hi! In the `FORWARD_OFFERED` case, we store `resolved_time` with NULL. There should handle this case.
BTW, may I ask you a question: what does `FORWARD_OFFERED` mean?

I'm trying to add local failcode #2524  , maybe I should add a new forward status: `FORWARD_FAILED_LOCAL` and also set `resolved_time` with NULL.

Reply:
`FORWARD_OFFERED` means that the forward was received, and potentially forwarded, but wasn't resolved yet. The state machine looks like this:

```
FORWARD_OFFERED --------> FORWARD_SETTLED
                                           \----> FORWARD_FAILED
```

`FORWARD_OFFERED` therefore is the initial state for all forwards, while `FORWARD_SETTLED` and `FORWARD_FAILED` are the terminal states for those. Hence we only set the `resolved_time` once we reach a terminal state.

Reply:
Thanks for your explain :)